### DISCUSSION TITLE
Group the tabs into work groups, setting the group color and name

### DISCUSSION THREAD

**Discussion by Angelk90** [zed-industries/zed#22068] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

<img width="1440" alt="Screenshot 2024-05-28 alle 14 19 55" src="https://github.com/zed-industries/zed/assets/20476002/f77120ba-d4e3-4c40-a3c7-4366e003cd13">
<img width="1006" alt="Screenshot 2024-05-28 alle 14 20 23" src="https://github.com/zed-industries/zed/assets/20476002/cfe7b0e9-1257-4cb8-aebe-255013a3da59">
<img width="374" alt="Screenshot 2024-05-28 alle 14 28 13" src="https://github.com/zed-industries/zed/assets/20476002/137c893d-c48c-42e3-9eeb-9b1a3afccf24">
<img width="837" alt="Screenshot 2024-05-28 alle 14 29 28" src="https://github.com/zed-industries/zed/assets/20476002/78466164-b407-4cbd-b545-f1c9a658cd16">
<img width="851" alt="Screenshot 2024-05-28 alle 14 28 05" src="https://github.com/zed-industries/zed/assets/20476002/889c25bd-2ac3-49e6-bad4-29951b076c1f">
<img width="1273" alt="Screenshot 2024-05-28 alle 14 27 54" src="https://github.com/zed-industries/zed/assets/20476002/6fb68d96-17f1-4482-bd18-d6d1e62131bc">

<img width="842" alt="Screenshot 2024-05-28 alle 14 31 48" src="https://github.com/zed-industries/zed/assets/20476002/5147e64b-b557-4447-be7e-fbcd044470c5">
<img width="845" alt="Screenshot 2024-05-28 alle 14 32 05" src="https://github.com/zed-industries/zed/assets/20476002/6b4fa20c-e93f-4ac6-b32d-aece04dfa221">

<img width="330" alt="Screenshot 2024-05-28 alle 14 34 45" src="https://github.com/zed-industries/zed/assets/20476002/dd9d6cb8-b073-4fbd-b3e8-7197f75bd9c1">

https://github.com/zed-industries/zed/assets/20476002/403f8ae2-827a-44f3-b170-784e0dd56834

**Comment by ghost** [comment]
I mean, in a code editor perspective, arent files already organized into different folders throughout the codebase? how would grouping in the tab bar be any different from picking a file from the explorer from an already grouped folder.

**Comment by Angelk90** [comment]
@evrsen : You may have the same file names `index.js, package.json, etc..` in both `server` and `frontend`, this may help you not to confuse them while you are working on both.

If I have to open the `index.js` file of `server`, I open the group it is in, work on it, and then close it, the tabs and the group remain there, if I still have to modify them, but the space I have for There is more tab space available and there is less chance of getting confused with all the open files.

The grouping is created at the moment, it is not based on how the files are set up in the folders, so you can also add files that do not belong to that folder, it is just a thing for you to help yourself.

I don't know if I explained myself.

**Comment by ghost** [comment]
Okay I get it, you explained great. Just FYI, you can have multiple files open with the same name and Zed will display it's parent folders name, so there is really no confusion about which index.js is which one when it comes to the tab bar.
![Screenshot 2024-05-28 at 16 27 54](https://github.com/zed-industries/zed/assets/146845123/a0885547-2cb8-4e04-8ad2-2c0ba1fdaeeb)

**Comment by Angelk90** [comment]
@evrsen : but don't you think that it is easier to see the color with the eye, instead of reading the name, you might even get confused or not even notice it.

**Comment by ghost** [comment]
Idk, this is also how VSCode does it.

**Comment by Angelk90** [comment]
@evrsen : It's not that you necessarily have to use it, if someone feels the need they can use it, it's just a proposal for something new that could be of help to someone.

For example, if there was something like Zed on I would use it, as I use it on Google Chrome as you can see from the images above.

On vscode there are extensions that allow you to group open tabs together and assign them a name.

**Comment by ghost** [comment]
Yeah I get that and can see how other people might find it useful, but in that case then maybe this should be an extension?
